Riot/Modules/Room/RoomViewController.m

```diff
@@ -7721,6 +7721,15 @@ - (void)documentPickerPresenter:(MXKDocumentPickerPresenter *)presenter didPickD
 {
     self.documentPickerPresenter = nil;
     
+    // Check maxUploadSize accepted by the home server before trying to upload.
+    NSUInteger maxUploadFileSize = self.roomDataSource.mxSession.maxUploadSize;
+    NSDictionary *fileAttributes = [NSFileManager.defaultManager attributesOfItemAtPath:url.path error:nil];
+    if (fileAttributes && fileAttributes.fileSize > maxUploadFileSize) {
```

Comment:
Have you checked if this takes any encryption overhead into account? Might be worth building a bit of leniency into this check.